# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de, 2024/04
# Allow to set AreTomo executable by environment variable WARPTOOLS_ARETOMO_EXE
diff -ru warp-1183a4a5ba8527987398b16d3d49ddb6e1bc9650/WarpTools/Commands/Tiltseries/AreTomoTiltseries.cs warp-1183a4a5ba8527987398b16d3d49ddb6e1bc9650_aretomo2/WarpTools/Commands/Tiltseries/AreTomoTiltseries.cs
--- warp-1183a4a5ba8527987398b16d3d49ddb6e1bc9650/WarpTools/Commands/Tiltseries/AreTomoTiltseries.cs	2024-04-29 18:44:27.000000000 +0200
+++ warp-1183a4a5ba8527987398b16d3d49ddb6e1bc9650_aretomo2/WarpTools/Commands/Tiltseries/AreTomoTiltseries.cs	2024-05-02 16:58:03.722779474 +0200
@@ -36,14 +36,19 @@
         [Option("delete_intermediate", HelpText = "Delete tilt series stacks generated for AreTomo")]
         public bool DeleteIntermediate { get; set; }
 
-        [Option("exe", Default = "AreTomo", HelpText = "Name of the AreTomo executable; must be in $PATH")]
+        [Option("exe", Default = "AreTomo", HelpText = "Name of the AreTomo executable; must be in $PATH. Affected by environment variable WARPTOOLS_ARETOMO_EXE.")]
         public string Executable { get; set; }
+        
     }
 
     class AreTomoTiltseries : BaseCommand
     {
         public override async Task Run(object options)
-        {
+        {  
+            bool has_exe =false; 
+            foreach (var arg in Environment.GetCommandLineArgs())  {if (arg.Contains("--exe")) {has_exe = true; break;}}; 
+            if (!has_exe && Environment.GetEnvironmentVariable("WARPTOOLS_ARETOMO_EXE")!=null )  
+                ((AreTomoTiltseriesOptions)options).Executable=Environment.GetEnvironmentVariable("WARPTOOLS_ARETOMO_EXE");
             await base.Run(options);
             AreTomoTiltseriesOptions CLI = options as AreTomoTiltseriesOptions;
             CLI.Evaluate();
